src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -6,55 +6,99 @@
 
 public class Range<T> implements Set<T> {
 
+    ArrayList<T> arrayList = new ArrayList<>();
+
     public int size() {
-        return 0;
+        return arrayList.size();
     }
 
     public boolean isEmpty() {
-        return false;
+
+        return arrayList.isEmpty();
     }
 
     public boolean contains(Object o) {
-        return false;
+        return arrayList.contains(o);
     }
 
     public Iterator<T> iterator() {
-        return null;
+        return arrayList.iterator();
     }
 
     public Object[] toArray() {
-        return new Object[0];
+
+        return arrayList.toArray();
     }
 
     public <T1> T1[] toArray(T1[] a) {
-        return null;
+
+        return (T1[]) arrayList.toArray();
     }
 
     public boolean add(T t) {
-        return false;
+        return  arrayList.add(t);
     }
 
     public boolean remove(Object o) {
-        return false;
+
+        return arrayList.remove(o);
     }
 
     public boolean containsAll(Collection<?> c) {
-        return false;
+        return arrayList.containsAll(c);
     }
 
     public boolean addAll(Collection<? extends T> c) {
-        return false;
+
+        return arrayList.addAll(c);
     }
 
     public boolean retainAll(Collection<?> c) {
-        return false;
+
+        return arrayList.retainAll(c);
     }
 
-    public boolean removeAll(Collection<?> c) {
-        return false;
+    public boolean removeAll(Collection<?> c)
+    {
+        return arrayList.removeAll(c);
     }
 
     public void clear() {
+        arrayList.clear();
+    }
+
+
+    public static Range<Integer> of(int i, int i1) {
+       Range<Integer> range= new Range<>();
+       if (i<i1) {
+           for (int j = i; j <= i1; j++) {
+               range.add(j);
+           }
+       }
+        return range;
+    }
+
+    public static Range<Float> of(float v, float v1) {
+
+        Range<Float> range= new Range<>();
+        if (v<v1) {
+            for (float j = v; j <= v1; j += 0.1f) {
+                range.add(j);
+            }
+        }
+        return range;
+
+
+    }
+
+    public static Range<Character> of(char a, char d, Function<Character, Character> function) {
```

Comment:
Need to support custom types

Reply:
public static <T extends Comparable> Range<T> of (T e1, T e2, Function<T, T> function) {
        Range<T> range = new Range<>();
        if (e1.compareTo(e2) < 0) {
            T i = e1;
            while (i.compareTo(e2) <= 0) {
                range.add(i);
                i = function.apply(i);
            }
        }
        return range;
    }

Reply:
Update PR, why do you check with if?

Reply:
And how do you use generic for static method? Run tests